### ISSUE TITLE
Error when using ggpredict() with plm() model

### ISSUE THREAD

**Issue by fschaffner** [strengejacke/ggeffects#121]
I get an error when calling ggpredict() after plm(). Reproducible example:

```{r}
library(ggeffects)
library(plm)

plm <- plm(mpg ~ cyl * disp,
          index = "am",
          data = mtcars,
          model = "within"
)

preds <- ggpredict(plm)
#> Error in crossprod(beta, t(X)): non-conformable arguments

sessionInfo()
#> R version 3.6.2 (2019-12-12)
#> Platform: x86_64-apple-darwin15.6.0 (64-bit)
#> Running under: macOS Catalina 10.15.2
#> 
#> Matrix products: default
#> BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
#> LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
#> 
#> locale:
#> [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> other attached packages:
#> [1] plm_2.2-3          ggeffects_0.14.1.1
#> 
#> loaded via a namespace (and not attached):
#>  [1] Rcpp_1.0.3       Formula_1.2-3    knitr_1.28       magrittr_1.5    
#>  [5] hms_0.5.3        MASS_7.3-51.5    insight_0.8.1    lattice_0.20-40 
#>  [9] rlang_0.4.5      bibtex_0.4.2.2   stringr_1.4.0    highr_0.8       
#> [13] tools_3.6.2      grid_3.6.2       nlme_3.1-144     xfun_0.12       
#> [17] miscTools_0.6-26 maxLik_1.3-8     htmltools_0.4.0  lmtest_0.9-37   
#> [21] yaml_2.2.1       digest_0.6.25    tibble_2.1.3     crayon_1.3.4    
#> [25] bdsmatrix_1.3-4  Rdpack_0.11-1    vctrs_0.2.3      gbRd_0.4-11     
#> [29] sjlabelled_1.1.3 evaluate_0.14    haven_2.2.0      rmarkdown_2.1.1 
#> [33] sandwich_2.5-1   stringi_1.4.6    compiler_3.6.2   pillar_1.4.3    
#> [37] forcats_0.5.0    zoo_1.8-7        pkgconfig_2.0.3
```

Thank you for looking into this!

**Comment by strengejacke** [comment]
Seems related to https://github.com/strengejacke/sjPlot/issues/594

I have looked into the code of `predict.plm()` and it indeed seems to be an issue with the plm package.

**Comment by nbrooks09** [comment]
This also seems to be a problem with the `lm_robust()` command within the `estimatr` library.

**Comment by annabokun** [comment]
Was this resolved by concluding that ggeffects is not compatible with plm?

**Comment by bckrick96** [comment]
I have the same question... how do we resolve predicting with plm?

**Comment by strengejacke** [comment]
As I mentioned [here](https://github.com/strengejacke/sjPlot/issues/594), I assume this is an issue of `predict.plm()`, so it's an issue of the *plm* package. At least the following example makes me think that the problem is located in the plm-package:

``` r
library(plm)
#> Warning: Paket 'plm' wurde unter R Version 4.0.3 erstellt
data("Produc", package="plm")
fit  <- plm(pcap ~ hwy + water + hwy*water, data=Produc, effect="twoway", index=c("state", "year"), model="within")

head(predict(fit))
#>   ALABAMA   ALABAMA   ALABAMA   ALABAMA   ALABAMA   ALABAMA 
#> 1330.2915 1135.8584 1009.8471  645.5366  324.9932  105.6784

nd <- data.frame(
  hwy = c(0, 10000, 20000, 0, 10000, 20000, 0, 10000, 20000),
  water = c(0, 0, 0, 4000, 4000, 4000, 8000, 8000, 8000)
)

predict(fit, newdata = nd)
#> Error in crossprod(beta, t(X)): non-conformable arguments
```

<sup>Created on 2020-11-15 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Maybe someone should ask the plm-maintainer how to resolve this problem?

**Comment by tappek** [comment]
There is now support for predict for plm's within models in the development version available here: https://github.com/ycroissant/plm

As the within case is special, one likely wants to input a pdata.frame and not a plain data frame; the help page ?predict.plm has the details.

**Comment by tappek** [comment]
For models produced by `plm::plm()`, there is a `predict` method available since `plm` version 2.6-2.

**Comment by strengejacke** [comment]
computing SE and CI for predictions still doesn't seem to be possible, because you can only compute the full model matrix, not for a specific data grid.
